src/client/hud.cpp

```diff
@@ -475,25 +475,6 @@ void Hud::drawHotbar(u16 playeritem) {
 				hotbar_itemcount / 2, mainlist, playeritem + 1, 0);
 		}
 	}
-
-	//////////////////////////// compatibility code to be removed //////////////
-	// this is ugly as hell but there's no other way to keep compatibility to
-	// old servers
-	if ((player->hud_flags & HUD_FLAG_HEALTHBAR_VISIBLE)) {
-		drawStatbar(v2s32(floor(0.5 * (float)m_screensize.X + 0.5),
-			floor(1 * (float) m_screensize.Y + 0.5)),
-			HUD_CORNER_UPPER, 0, "heart.png",
```

Comment:
ID 0 is used here; might be the reason why the HUDs sometimes freak out and cannot be changed when they got this ID. Removing this code is highly recommended.